apps/api/src/workspace/service/workspace.service.ts

```diff
@@ -844,18 +840,137 @@ export class WorkspaceService {
     return data
   }
   
-  async globalSearch(workspaceId: string, searchTerm: string, user: User) {
-    const projects = await this.projectService.getProjectsOfWorkspace(user, workspaceId, searchTerm);
-    const environments = await this.environmentService.getEnvironmentsOfWorkspace(user, workspaceId, searchTerm);
-    const secrets = await this.secretService.getSecretsOfWorkspace(user, workspaceId, searchTerm);
-    const variables = await this.variableService.getVariablesOfWorkspace(user, workspaceId, searchTerm);
-
-    return {
-      projects,
-      environments,
-      secrets,
-      variables,
-    };
+  async globalSearch(
+    user: User,
+    workspaceId: string,
+    searchTerm: string
+  ): Promise<{
+    projects: Project[];
+    environments: Environment[];
+    secrets: Secret[];
+    variables: Variable[];
+  }> {
+    // Check authority over workspace
+    await this.authorityCheckerService.checkAuthorityOverWorkspace({
+      userId: user.id,
+      entity: { id: workspaceId },
+      authority: Authority.READ_WORKSPACE,
+      prisma: this.prisma,
+    });
+
+    // Query all entities based on the search term and permissions
```

Comment:
You would first need to compute all the projects that the user has access to in the workspace. Then, you would need to pass that `projectId` array to each and every query method. 

Next up, you would need to check if the entity contains `some` of the project ids:
![image](https://github.com/keyshade-xyz/keyshade/assets/83924254/f2d2eac9-ff5f-4014-9b9a-4e436c5f9c52)

Only that in your case, you would need to specify `in` and not `notIn`. Also, que